tests/conftest.py

```diff
@@ -89,7 +89,7 @@ def session_setup_teardown():
 
     if clean:
         restore_db_from_snapshot()
-    shutil.rmtree(snapshot_dir)
```

Comment:
I think we should raise this error. 
It is good to know that we havent closed all files properly.

Reply:
sure, but then I'm gonna have to hunt down some errors, because this happens for me every local test run. I can at least take it out of the PR if you want

Reply:
Yeah, I have tried to reduce this stuff as much as possible, but hydromt adds a fileHandler to the logger if it doesnt have one (in setuplog) and then doesnt close it when errors are raised. I tried fixing this with the FloodAdaptLogging class, and the __del__ function for the SfincsAdapter and FiatAdapter, but it has never really been 100% when other stuff is also breaking.
Fixing this was quite time consuming so when I had a mostly working solution, I just left it